### PR TITLE
[test] Make dependency check exceptions more granular

### PR DESCRIPTION
**Pull request by saimidu**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
- Make dependency check test exceptions for false-positives more granular
- Add exception for MXNet 1.8 Neuron

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [x] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]

#### EIA/NEURON Testing Checklist
* When creating a PR:
- [x] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true` or `neuron_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by arjkesh** [comment]
Nit: I'd change `exceptions` to `allowlist` or something similar in the PR description  - the term exceptions is a bit misleading

File: test/dlc_tests/sanity/test_pre_release.py [hunk 1]
```diff
@@ -249,14 +249,20 @@ def _run_dependency_check_test(image, ec2_connection, processor):
...
 
     allow_openssl_cve_fw_versions = {
-        "tensorflow": ["1.15", "2.3", "2.4", "2.5", "2.6"],
-        "mxnet": ["1.9"],
-        "pytorch": [],
-        "huggingface_pytorch": ["1.8"],
-        "huggingface_tensorflow": ["2.4"]
+        "tensorflow": {
+            "1.15": ["cpu", "gpu", "neuron", "eia"],
+            "2.3": ["cpu", "gpu", "eia"],
```
Comment (arjkesh) [inline test/dlc_tests/sanity/test_pre_release.py:254]: have you verified the openssl version in the EI/Neuron containers for these versions? This should be done prior to merge

Comment (saimidu) [reply test/dlc_tests/sanity/test_pre_release.py:254]: I've added exceptions for whatever was previously excepted. Will double check these to make sure.

File: test/dlc_tests/sanity/test_pre_release.py [hunk 2]
```diff
@@ -249,14 +249,20 @@ def _run_dependency_check_test(image, ec2_connection, processor):
     short_fw_version = re.search(r"(\d+\.\d+)", image).group(1)
 
     allow_openssl_cve_fw_versions = {
-        "tensorflow": ["1.15", "2.3", "2.4", "2.5", "2.6"],
-        "mxnet": ["1.9"],
-        "pytorch": [],
-        "huggingface_pytorch": ["1.8"],
-        "huggingface_tensorflow": ["2.4"]
+        "tensorflow": {
+            "1.15": ["cpu", "gpu", "neuron", "eia"],
```
Comment (arjkesh) [inline test/dlc_tests/sanity/test_pre_release.py:253]: processor seems to come from the test directly. Moreover, in this script, it was just used to name the container it seems. I don't believe that we have a dependency check specifically for `eia` - you may need to update `test_dependency_check_cpu` if you need to handle EI separately.
